modules/network/src/integrationTest/java/org/apache/ignite/internal/network/ItStaticNodeFinderTest.java

```diff
@@ -53,12 +55,27 @@ protected boolean needInitializeCluster() {
 
     @Test
     void testNodeShutdownOnNodeFinderFailure(TestInfo testInfo) {
-        Throwable throwable = assertThrowsWithCause(
-                () -> CLUSTER.startAndInit(testInfo, initialNodes(), cmgMetastoreNodes(), this::configureInitParameters),
-                IgniteInternalException.class);
+        LogInspector logInspector = new LogInspector(FailureManager.class.getName());
 
-        IgniteInternalException actual = (IgniteInternalException) unwrapRootCause(throwable);
-        Assertions.assertEquals(ADDRESS_UNRESOLVED_ERR, actual.code());
-        Assertions.assertEquals("No network address found", actual.getMessage());
+        logInspector.addHandler(
+                evt -> {
+                    if (evt.getLevel() == Level.ERROR) {
+                        IgniteInternalException actual = (IgniteInternalException) unwrapRootCause(evt.getThrown());
+                        assertEquals(ADDRESS_UNRESOLVED_ERR, actual.code());
+                        assertEquals("No network addresses found", actual.getMessage());
+
+                        return true;
+                    }
+
+                    return false;
+                }, () -> assertThrows(IllegalStateException.class, CLUSTER::aliveNode));
+
+        logInspector.start();
+
+        try {
+            CLUSTER.startAndInit(testInfo, initialNodes(), cmgMetastoreNodes(), this::configureInitParameters);
```

Comment:
So we are not throwing the exception, right? If it's an embedded mode, it would make sense to throw it to make it more visible to the application. If it's not, an exception thrown here will prevent the node start. So we probably don't need any failure manager interaction, after all...